pkg/component/component.go

```diff
@@ -60,21 +60,39 @@ func CreateFromDir(client *occlient.Client, name string, ctype string, dir strin
 
 // Delete whole component
 func Delete(client *occlient.Client, name string, applicationName string, projectName string) (string, error) {
+
 	cfg, err := config.New()
 	if err != nil {
-		return "", errors.Wrapf(err, "unable to delete component %s", name)
+		return "", errors.Wrapf(err, "unable to create new configuration to delete %s", name)
 	}
 
 	labels := componentlabels.GetLabels(name, applicationName, false)
 
 	output, err := client.Delete("all", "", labels)
 	if err != nil {
-		return "", errors.Wrapf(err, "unable to delete component %s", name)
+		return "", errors.Wrapf(err, "error deleting component %s", name)
 	}
 
-	err = cfg.SetActiveComponent("", projectName, applicationName)
+	// If there are components that still exist (len > 0) then set it to the default component..
+	components, err := List(client, applicationName, projectName)
 	if err != nil {
-		return "", errors.Wrapf(err, "unable to delete component %s", name)
+		return "", errors.Wrapf(err, "unable to retrieve list of components")
+	}
+
+	// If there's more than one component, set it to the first one..
+	if len(components) > 0 {
+		err = cfg.SetActiveComponent(components[0].Name, applicationName, projectName)
+		if err != nil {
+			log.Warnf("No active component found, setting active component to %s", components[0].Name)
+			return "", errors.Wrapf(err, "unable to set current component %s", name)
+		}
+	} else {
+		// Unset to blank
+		err = cfg.UnsetActiveComponent(applicationName, projectName)
+		if err != nil {
+			return "", errors.Wrapf(err, "error unsetting current component while deleting %s", name)
+		}
+		log.Warnf("No default component has been set")
```

Comment:
I don't think this is Warn. It is the normal state. There are just no components left. 
I think that the message should say just that.